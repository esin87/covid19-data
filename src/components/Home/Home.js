import React from 'react';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Button from 'react-bootstrap/Button';
import { LinkContainer } from 'react-router-bootstrap';
const Home = () => {
	return (
		<Jumbotron>
			<h1>Covid 19 Data</h1>
			<p>
				Search for Covid-19 case information by country. Data courtesy of{' '}
				<a
					href='https://covid19api.com/'
					target='_blank'
					rel='noopener noreferrer'>
					covid19api.com
				</a>
			</p>
			<p>
				<LinkContainer to='/search'>
					<Button variant='primary'>Start searching</Button>
				</LinkContainer>
			</p>
		</Jumbotron>
	);
};

export default Home;
