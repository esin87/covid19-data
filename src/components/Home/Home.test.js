import React from 'react';
import Home from './Home';
import { mount, render } from 'enzyme';
import { HashRouter } from 'react-router-dom';

describe('Home component', () => {
	let component;
	beforeEach(() => {
		component = mount(
			<HashRouter>
				<Home />
			</HashRouter>
		);
	});

	it('renders', () => {
		render(component);
	});

	it('should match the snapshot', () => {
		expect(component.html()).toMatchSnapshot();
	});

	it('contains a h1 element with the appropriate text', () => {
		expect(component.find('h1').text()).toBe('COVID-19 Data');
	});

	it('contains a first anchor element with the appropriate text', () => {
		expect(component.find('a').at(0).text()).toBe('covid19api.com');
	});

	it('contains a second anchor element with the appropriate text', () => {
		expect(component.find('a').at(1).text()).toBe('Start searching');
	});
});
