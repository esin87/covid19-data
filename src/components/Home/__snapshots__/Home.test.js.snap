// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match the snapshot 1`] = `"<div class=\\"jumbotron\\"><h1>COVID-19 Data</h1><p>A free, open-source tool to search for up-to-date COVID-19 case information by country. </p><p>Data courtesy of <a href=\\"https://covid19api.com/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">covid19api.com</a>.</p><p><a href=\\"#/countries\\" class=\\"btn btn-primary\\">Start searching</a></p></div>"`;
